tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/dialogs/UIEditWorkingSetWizardAuto.java

```diff
@@ -119,4 +146,17 @@ public void testEditPage() throws Throwable {
 
 		DialogCheck.assertDialogTexts(getWizardDialog());
 	}
+
+	@Override
+	public void doSetUp() throws Exception {
+		super.doSetUp();
+		setAutoBuilding(false);
+	}
+
+	@Override
+	public void doTearDown() throws Exception {
+		super.doTearDown();
+		setAutoBuilding(true);
```

Comment:
You shouldn't assume `true` was the original value. 

Instead, save the original value in `doSetUp` and apply it again in `doTearDown`

Reply:
Since there will be other tests that do not clean up properly, it is usually better to restore defaults than restoring the state before the test. So a general cleanup (also used by the platform resource tests), is to just restore the default workspace description:
```java
ResourcesPlugin.getWorkspace().setDescription(Workspace.defaultWorkspaceDescription());
```

Reply:
This would make the test(s) meddle with each other but I guess since one is restoring the **default** value then it should be OK.

In that case, why not push this all the way to the top to `UITestCase::tearDown`?

Reply:
> This would make the test(s) meddle with each other

It would do the exact opposite.

> In that case, why not push this all the way to the top to `UITestCase::tearDown`?

Because it is out-of-scope of this PR to refactor the complete UI test infrastructure. Having unified cleanups to ensure proper test isolation would of course make sense but should be done (1) as a separate task and (2) should be done properly using mechanisms like JUnit 4 rules or JUnit 5 extensions instead of fostering test class inheritance.

Reply:
> > This would make the test(s) meddle with each other
> 
> It would do the exact opposite.

It's a hack that makes this test "fix whatever other tests might have broken", that's what I meant. But it's ok, it's outside the scope of this PR to hunt down all the offending tests ;-) . BTW do you already know of any such tests (by name) or does this have to be investigated yet?

Reply:
> BTW do you already know of any such tests (by name) or does this have to be investigated yet?

If I knew them, I would have created a ticket or fixed them. Most of the test failures I faced in the last year have one of two reasons:
1. Race conditions
2. Improper test isolation

What can be done systematically to ensure isolation is to encapsulate state changes or potential leaks (like job execution) into proper rules (JUnit 4) or extensions (JUnit 5).